client/app/services/file-manager.js

```diff
@@ -59,6 +54,7 @@ export default class FileManager {
   }
 
   uploadFiles() {
+    this.filesToUpload.files.forEach((file) => this.existingFiles.pushObject(file));
```

Comment:
Nice!

What if the file uploads fail? What if we `await` the Promse.all on the line below, wrapped in a try / catch, then say:

```
this.existingFiles.pushObjects(this.filesToUpload.files)
```